### PR TITLE
IDL: add information about cluster instantiations on endpoints

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
IDL not showing what clusters are available on each endpoint.

#### Change overview
Add endpoint query functions in helpers and use those in templates.

NOTE: I did not find how the existing code does this ... maybe (I really hope so!) there is a better way.

#### Testing
Looked at ZAP output. Only IDL was changed.

### CODE REVIEW

File: examples/tv-casting-app/tv-casting-common/tv-casting-app.matter [hunk 1]
```diff
@@ -2459,11 +2459,66 @@ server cluster WindowCovering = 258 {
...
+  instantiate cluster OtaSoftwareUpdateProvider;
+  instantiate cluster OperationalCredentials;
+  instantiate cluster RelativeHumidityMeasurement;
+  instantiate cluster SoftwareDiagnostics;
+  instantiate cluster ThreadNetworkDiagnostics;
+  instantiate cluster UserLabel;
+  instantiate cluster WiFiNetworkDiagnostics;
 }
 
 endpoint 1 {
```
Comment (andy31415) [inline examples/tv-casting-app/tv-casting-common/tv-casting-app.matter:2483]: @chrisdecenzo - are these correct? Why does a TV casting app have things like DoorLock or PressureMeasurement on endpoints?

Comment (andy31415) [reply examples/tv-casting-app/tv-casting-common/tv-casting-app.matter:2483]: Note that this has nothing to do with this PR ... but this PR exposes what is in a ZAP file in a more readable way. Or at least it tries to - if it is  wrong please let me know.

File: src/app/zap-templates/templates/app/MatterIDL.zapt [hunk 2]
```diff
@@ -79,9 +79,11 @@
 
 {{/all_user_clusters}}
 
-{{#user_endpoints}}
+{{#chip_endpoints}}
 endpoint {{endpointId}} {
-    {{!TODO: report device types and cluster instantionation. }}
+  {{#chip_endpoint_clusters}}
+  instantiate cluster {{asUpperCamelCase name}};
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/templates/app/MatterIDL.zapt:85]:
```suggestion
  instantiate {{side}} cluster {{asUpperCamelCase name}};
```
in that it's worth telling apart servers and clients, I suspect.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/templates/app/MatterIDL.zapt (hunk 2 above)
Comment (andy31415) [reply src/app/zap-templates/templates/app/MatterIDL.zapt:85]: Do they have any meaning here?

I thought our endpoint info is always server-side: it means what my device exposes to the  world.
When I create a client, I will not be able to tell on which  endpoint a lock/light/other actually resides, so saying "endpoint 5 { instantiate client cluster Lock }" seems odd to me. If I need to control a light, I should query the remote  side for what endpoints it exposes.

Comment (andy31415) [reply src/app/zap-templates/templates/app/MatterIDL.zapt:85]: I will follow up offline and do this in a followup if it is useful for understanding clusters.

**Pull request merged**